core/src/cluster_slots.rs

```diff
@@ -54,13 +58,20 @@ impl ClusterSlots {
                 })
                 .collect()
         };
+        // Discard slots at or before current root or epochs ahead.
+        let slot_range = (root + 1)
```

Comment:
hmmm I think we need some buffer for the slots before the root for nodes that are catching up. There's also a use case currently for handling duplicate slots where nodes that have gotten the wrong version of a block rely on seeing `EpochSlots` reach a certain percentage of the threshold before initiating the duplicate repair protocol. cc @AshwinSekar 

Is it feasible to keep around at least the latest epoch of `EpochSlots`?

Reply:
That sounds like a separate change independent of this commit.
Current master code is already discarding epoch slots at or before current root:
https://github.com/solana-labs/solana/blob/692aa9914/core/src/cluster_slots.rs#L61-L63
https://github.com/solana-labs/solana/blob/692aa9914/core/src/cluster_slots.rs#L82

Reply:
oh shoot this is the internal update for *other* nodes' epoch slots wer're receiving, but we're still persisting our own updates that are earlier than the root. Sorry for the confusion!